### PR TITLE
Add LoongArch CPU support. Allow building kodi on LoongArch Linux system

### PR DESCRIPTION
**Pull request by phorcys**
Add LoongArch CPU support.
## Description
LoongArch is a new RISC architecture made by Loongson. 
This pull request add basic LoongArch support.

Patch tested on a gentoo linux system with kernel 6.0 glibc2.36

### CODE REVIEW

**Review by wsnipex — APPROVED** [review]

**Review by fuzzard — CHANGES_REQUESTED** [review]
Can you sort the new defines alphabetical.

They were reordered not that long ago, so lets try and keep them in order still.

**Comment by phorcys** [comment]
> Can you sort the new defines alphabetical.
> 
> They were reordered not that long ago, so lets try and keep them in order still.

Sorry, my fault.
Fixed the alphabetical problem.

**Review by fuzzard — APPROVED** [review]

File: xbmc/utils/MathUtils.h [hunk 1]
```diff
@@ -27,6 +27,7 @@
     defined(__alpha__) || \
     defined(__arc__) || \
     defined(__arm__) || \
+    defined(__loongarch) || \
```
Comment (xen0n) [inline xbmc/utils/MathUtils.h:30]: Should be `__loongarch__` instead... Mind sending a fix PR?

Comment (phorcys) [reply xbmc/utils/MathUtils.h:30]: > Should be `__loongarch__` instead... Mind sending a fix PR?
Yes, an typo.

Comment (howie-f) [reply xbmc/utils/MathUtils.h:30]: https://github.com/xbmc/xbmc/pull/22073

**Pull request merged**